.werft/util/werft.ts

```diff
@@ -86,27 +108,48 @@ export class Werft {
     }
 
     public done(slice: string) {
-        const span = this.sliceSpans[slice]
-        if (span) {
-            span.end()
-            delete this.sliceSpans[slice]
+
+        // This is only for backwards compatibility!
+        if (!this.phases[this.currentPhase].slices[slice]) {
+            this.newSlice(slice);
+        }
+
+        if (this.phases[this.currentPhase].slices[slice].closed) {
+            console.log(`WARNING: The slice "${slice}" has already been closed!`);
+        } else {
+            this.phases[this.currentPhase].slices[slice].span.end();
+            this.phases[this.currentPhase].slices[slice].closed = true;
+            console.log(`[${slice}|DONE]`)
         }
-        console.log(`[${slice}|DONE]`)
     }
 
     public result(description: string, channel: string, value: string) {
         exec(`werft log result -d "${description}" -c "${channel}" ${value}`);
     }
 
-    private endPhase() {
-        // End all open slices
-        Object.entries(this.sliceSpans).forEach((kv) => {
-            const [id, span] = kv
-            span.end()
-            delete this.sliceSpans[id]
-        })
+    public endPhase(name?: string) {
```

Comment:
What's the use case for supporting `name` as a parameter? 

It seems wrong because if I'm not mistaken only on phase can be open at a given time. This implies that `this.currentPhase` is always the phase that's being closed.

Reply:
Adding the name explicitly would make the code more readable and help do understand better to which phase it belongs.